src/h.ts

```diff
@@ -17,29 +17,48 @@ function addNS(data: any, children: VNodes | undefined, sel: string | undefined)
   }
 }
 
+function flattenChildren(children: VNodeChildren[], flattened: VNode[]) {
+  for (const child of children) {
+    if (child !== undefined && child !== null && child !== false) {
+      if (is.array(child)) {
+        flattenChildren(child, flattened);
+      } else if (is.primitive(child)) {
+        flattened.push(vnode(undefined, undefined, undefined, String(child), undefined));
+      } else {
+        flattened.push(child);
+      }
+    }
+  }
+  return flattened;
+}
+
 export function h(sel: string): VNode;
 export function h(sel: string, data: VNodeData): VNode;
 export function h(sel: string, children: VNodeChildren): VNode;
-export function h(sel: string, data: VNodeData, children: VNodeChildren): VNode;
+// jsx/tsx compatible h signature
+export function h(sel: string | Function, data: VNodeData | null, ...children: VNodeChildren[]): VNode;
 export function h(sel: any, b?: any, c?: any): VNode {
-  var data: VNodeData = {}, children: any, text: any, i: number;
+  var data: VNodeData = {}, children: any, text: any;
   if (c !== undefined) {
-    data = b;
+    if (arguments.length > 3) {
+      c = Array.prototype.slice.call(arguments, 2);
+    }
+    if (b !== null) data = b;
     if (is.array(c)) { children = c; }
     else if (is.primitive(c)) { text = c; }
     else if (c && c.sel) { children = [c]; }
-  } else if (b !== undefined) {
+  } else if (b !== undefined && b !== null) {
     if (is.array(b)) { children = b; }
     else if (is.primitive(b)) { text = b; }
     else if (b && b.sel) { children = [b]; }
     else { data = b; }
   }
   if (children !== undefined) {
-    for (i = 0; i < children.length; ++i) {
-      if (is.primitive(children[i])) children[i] = vnode(undefined, undefined, undefined, children[i], undefined);
-    }
+    children = flattenChildren(children, []);
```

Comment:
Potential performance regression?

Reply:
Potential, I haven't benchmarked it. Does snabbdom have any benchmarking tests/utils for this. I can do a benchmark. I don't think it should be a major regression though compared to tree diffing which is a much bigger cost. Agreed though, best to benchmark and have conclusive results.

Reply:
There seems to be a benchmark [here](https://github.com/snabbdom/snabbdom/tree/master/perf). Not sure whether it covers this change. If it doesn't, would you mind adding that coverage, please?

Reply:
Thinking through this I had an idea. What if we don't bundle everything in h. Expose a different function `jsx` and have h only handle the null bit. 

This not only alleviates the perf concerns, it moves towards a design of composition